### ISSUE TITLE
feat[dev]: ✨ implement A11y improvements to sd-carousel & sd-carousel-item

### ISSUE THREAD

**Issue by yoezlem** [solid-design-system/solid#1470]
## Description
This task describes the Storybook implementation for the sd-brandshape component as part of the A11y improvements outlined in Epic https://github.com/solid-design-system/solid/issues/1465.

A11y improvements for sd-sd-carousel and sd-carousel-item can be found here: [sd-carousel Notes](https://www.notion.so/matuzo/sd-carousel-sd-carousel-item-105f3f6e1bc080c3af6cf953f669bb88).

----------------------------------------

## Comment
**Problem:** There’s no announcement when slides change because #scroll-container is not a live region. aria-atomic and aria-busy alone do not affect the #scroll-container. It must be a live region.
**Solution:** Add aria-live=polite|assertive to the #scroll-container or use role=status|alert instead.

**Problem:** #scroll-container is a generic element (div) but is focusable. If you add tabindex=0 to an element, it must also have an accessible name and a non-generic role.
**Solution:** Use role=status instead of aria-atomic=true. It’s a shorthand for aria-live=polite and aria-atomic=true. It gives you the setting you want and a semantic role.

**Problem:** Manually turning #scroll-container into a live region, you get an announcement, but it’s broken. You get something like “Previous slide NaN / 3 Next slide [alt]”.
**Solution:** When you add the role also fix and simplify the announcement, e.g. “[alt] 2 of 3”.

**Problem:** The play/pause button doesn't indicate whether the carousel is currently playing.
**Solution:** By default, set aria-pressed="true" and toggle the value when the button is activated. Tigh this to the function which stops or plays the carousel.

**Problem:** The play/pause button has no accessible name.
**Solution:** Add one.

**Problem:** Focus is lost on the previous/next buttons when you reach the first/last slide because the buttons turn into the disabled state.
**Solution:** Move focus to the other active button or change your solution for disabled buttons (see [@sd-button](https://www.notion.so/matuzo/sd-button-104f3f6e1bc0800b9f72d2a046316ec3)).

## Notes
- There are options to change the previous and next icons, but is it also possible to change the accessible names, e.g., provided by aria-label? The would be important for i18n.
- As the role / accessible name should not be changed but the icons need to be flexible (skinning), we should only make this clear in the documentation.

## Best Practices
Carousels are generally hard for screen reader users to use; try to avoid it.

## Related WCAG criteria
- https://www.w3.org/WAI/WCAG22/quickref/#name-role-value
- https://www.w3.org/WAI/WCAG22/quickref/#status-messages

## Additional resources
- https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions

### Open Questions
- [ ] Question1
- [ ] Question2

## DoR
- [x] Item has been estimated by the team
- [x] Item dependencies have been identified and documented

## DoD
- [ ] Documentation has been created/updated (if applicable)
- [ ] Migration Guide has been created/updated (if applicable)
- [ ] Relevant E2E tests (Features, A11y, Bug fixes) are created/updated
- [ ] Relevant stories (Features, A11y) are created/updated
- [ ] Implementation works successfully on `feature` branch
